django/contrib/admin/helpers.py

```diff
@@ -225,44 +225,53 @@ class InlineAdminFormSet:
     A wrapper around an inline formset for use in the admin system.
     """
     def __init__(self, inline, formset, fieldsets, prepopulated_fields=None,
-                 readonly_fields=None, model_admin=None):
+                 readonly_fields=None, uneditable_fields=None, has_add_permission=True,
+                 has_change_permission=True, has_delete_permission=True, model_admin=None):
```

Comment:
Should we add `has_view_permission` for consistency here? I'm not sure I like that we assume that users have these permissions by default. Should these not be added after `model_admin` to not break backwards compatibility of parameter ordering? Currently only `has_add_permission` gets used in this class and all three of these flags are used at the same time outside of the class definition. Would it make more sense to pass in a `view_only=True` or `editable=False` flag if we do not need the granularity?

As previously mentioned, I think that the overlap between `readonly_fields` and `uneditable_fields` is messy. I don't know whether we can make this work with `readonly_fields` alone or whether we need to come up with another solution.